### PR TITLE
[F11-3] TaskLinker

### PR DESCRIPTION
**Pull request by wendelinwemhoener**
TaskLinker helps students using flashcards for memorizing course material to create, review and schedule their flashcards.

### CODE REVIEW

File: src/main/java/seedu/duke/flashcard/FlashcardStorage.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
+package seedu.duke.flashcard;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.util.logging.*;
```
Comment (madhanse) [inline src/main/java/seedu/duke/flashcard/FlashcardStorage.java]: Please change this import because wildcard imports are not allowed

**Comment by madhanse** [comment]
Hi all, please make sure to add whitespaces in if-else statements and functions as per the guidelines, and avoid wildcard imports. To improve code quality, try to remove unnecessary blank spaces. Also, please pass the Gradle tests before pushing your commits for good practice.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -8,22 +8,61 @@
...
+| v2.0    | new user                  | be able to invoke a help command                               | get advice/tips for how to use TaskLinker if I don' know how                                  |
+| v2.0    | user reviewing flashcards | rate how hard it was to remember their content                 | see harder flashcards more often and easy flashcards less frequently during reviews           |
+| v2.0    | experienced user          | use the delete and review feature by typing a one-line command | be faster than having to go through the prompt-answer workflow these features usually require |
+| v2.0    | user                      | find events from my Calendar                                   | find events without listing them all                                                          |
+| v2.0    | user                      | delete all events from my Calendar                             | don't need to remove events one by one                                                        |
 
 ## Non-Functional Requirements
```
Comment (martinschnder) [inline docs/DeveloperGuide.md]: This part is missing

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -8,22 +8,61 @@
...
```
Comment (martinschnder) [inline docs/DeveloperGuide.md:428]: The glossary and instructions are missing as well

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -8,22 +8,61 @@
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Storage Components
```
Comment (martinschnder) [inline docs/DeveloperGuide.md]: There is currently no diagram in the document

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -8,22 +8,61 @@
...
 
 ## User Stories
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| Version | As a ...                  | I want to ...                                                  | So that I can ...                                                                             |
+|---------|---------------------------|----------------------------------------------------------------|-----------------------------------------------------------------------------------------------|
+| v1.0    | user                      | Delete event from my Calendar                                  | remove unnecessary events from the Calendar                                                   |
```
Comment (martinschnder) [inline docs/DeveloperGuide.md:283]: the users stories are clear and quite complete

**Comment by madhanse** [comment]
image

Non-functional requirements and glossary are missing

**Comment by madhanse** [comment]
image
The developer guide looks incomplete as it is missing UML diagrams and there are no details about the class structure.
Please follow the template of AB3 developer guide

**Comment by madhanse** [comment]
image
Some details can be omitted, as it is quite obvious that it is a constructor.

**Comment by madhanse** [comment]
image
Need to use same line for alternative paths. Cannot use duplicate objects

File: docs/photo/FlashcardStorage-0.png
Comment (ziyi105) [inline docs/photo/FlashcardStorage-0.png:1]: Would the "new" call by FlashCardComponent on FlashCardDirectory be a bit confusing? if it is a constructor call perhaps you could use `new FllashCardDirectory()` which would be clearer?

File: docs/UserGuide.md
Comment (ziyi105) [inline docs/UserGuide.md:1]: Perhaps you could insert a table of contents here? You can generate it easily with intellij generate feature!

File: docs/UserGuide.md [hunk 6]
```diff
@@ -2,41 +2,433 @@
...
+##### Command format and example of usage in beginner mode
+
+Command Format: `review flashcards`
+
+Example of usage:
+<pre>
+Enter your command: <b>review flashcards</b>
+    How do you want to review your flashcards?
+        a) random mode
+        b) spaced repetition mode
```
Comment (ziyi105) [inline docs/UserGuide.md:206]: Perhaps you could put them as code block?

**Pull request closed without merging**